### DISCUSSION TITLE
Google Summer of Code 2021 application

### DISCUSSION THREAD

**Discussion by GMishx** [fossology/fossology#1881] [Ideas]
Hello all! FOSSology is planning to apply for [Google Summer of Code](https://summerofcode.withgoogle.com) this year as well. In preparation to do so, we have created a Wiki page allowing a quick overview to FOSSology and proposals. https://github.com/fossology/fossology/wiki/Google-Summer-of-Code-Proposals-2021

I am opening this discussion here to allow quick contact for anyone interested in the program. Please feel free to drop any questions here (or start a new discussion) you have regarding the program, you want to submit a proposal idea, you want to be mentor in the program, etc.

**Comment by NicolasToussaint** [comment]
Great !

I would like to propose my participatation in mentoring the REUSE.Software [project](https://github.com/fossology/fossology/wiki/Google-Summer-of-Code-Proposals-2021#integrating-reuse).

For information, the following issues already exist
- _Integrate REUSE standard_: #1592
- _Support the `SPDX-FileCopyrightText` keyword explicitly in the copyright agent_ #1513 
- _Detect 'foobar.license' files, as part of RESUE.Software standard_ #1833
    - We have already started working on that one

**Reply by GMishx** [reply]
That is great Nicolas. I have added your name in the mentor list and added the list of issues to the proposal.

**Comment by sjha2048** [comment]
Last year I applied for [Rewriting the UI](https://github.com/fossology/fossology/wiki/Google-Summer-of-Code-Proposals-2021#rewrite-fossology-ui-for-example-using-angular-bootstrap-vuejs-or-reactjs) project and gained some valuable insights, unfortunately, I was not able to continue this project, this year I am hoping to mentor for the same project and share my insights with you all.

[Here](https://xd.adobe.com/view/5a36e175-6496-48d7-4b9b-3221c85df525-2c07/) is a reference design that I proposed last year.
for further dive, you can check out this pull request #1774

**Reply by GMishx** [reply]
Hello @sjha2048 , I have added you to the mentor list and provided the link to your design.

**Comment by hastagAB** [comment]
Hello,

I would like to propose one project idea for GSoC 2021: **Generating OSS License dataset**

Goal: To create/generate OSS License dataset for [Atarashi](https://github.com/fossology/atarashi)

Description: To implement any Machine learning/Deep learning algorithm we need a better and bigger dataset of [SPDX Licences](https://spdx.org/licenses/). But unfortunately, there exists no such dataset for open source licenses on the web.

There is a loose implementation of n-gramming different permutations and combinations of license text paragraphs. Ref: [SPDX-OSS Dataset](https://github.com/hastagAB/SPDX-OSS-Dataset). This method needs further improvement to make the dataset more accurate and realistic. 

Few suggested improvements: 
- Shifting from txt files to SPDX JSON endpoint
- Differentiating License Header from Full Text
- Adding FOSSology Nomos agent [STRINGS.in](https://github.com/fossology/fossology/blob/master/src/nomos/agent/STRINGS.in) regex in dataset creation
- Apart from these we strongly encourage students to pitch in some new ideas/methods/algorithms to achieve more accurate results. 

Resources: 
- [Atarashi](https://github.com/fossology/atarashi)
- [SPDX Licences](https://spdx.org/licenses/)
- [SPDX-OSS Dataset](https://github.com/hastagAB/SPDX-OSS-Dataset)
- [STRINGS.in](https://github.com/fossology/fossology/blob/master/src/nomos/agent/STRINGS.in)

**Reply by GMishx** [reply]
Thanks for great suggestion Ayush. I have added it to the Wiki: https://github.com/fossology/fossology/wiki/Google-Summer-of-Code-Proposals-2021#generating-oss-license-dataset

**Comment by ayan-biswas0412** [comment]
Hello I am Ayan Biswas , Currently pursuing 2nd year undergrad in Electronics and Telecommunication Engineering from Jadavpur University , I am constantly working digital and analog electronics simulation for about 1.5 years and  i am an experienced MERN and MEAN stack developer . I am an aspirant for the GSOC 2021 in the **UI-enhancement project** , can anyone guide me what are the initial works available to get accustomed to this organisation for the upcoming GSOC or with whom I may discuss about that particular project. Thanking you all in advance

**Reply by GMishx** [reply]
Hello Ayan, glad to have you here.
You can install FOSSology and checkout the existing UI.
You can refer to the UI provided in the project description and comeup with some more interesting ideas.
You can checkout some issues related to UI from https://github.com/fossology/fossology/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22
And please feel free to drop a message in Slack :+1:

**Reply by ayan-biswas0412** [reply]
yeah I've messaged you in slack

**Comment by GMishx** [comment]
Since GSoC'21 results are announced yesterday, we can close this conversation. But I can not find a way to do so :sweat_smile:
